assets/js/hooks/cell.js

```diff
@@ -211,6 +211,15 @@ const Cell = {
         // gives it focus
         if (!this.isFocused || !this.insertMode) {
           this.currentEditor().blur();
+        } else if (this.insertMode) {
+          const lineNumber = this.currentEditor().getLineNumberAtCursor();
```

Comment:
Imagine this scenario:

1. Enter insert mode, line 1 is focused
2. Move down to line 3
3. Go to another cell
4. Go back

I would expect to go back to line 3, but we only register the navigation in step 1.

What we actually want is to subscribe to cursor position changes, send that to the session. If the cell id is the same as last entry, we only update the line, otherwise we can push a new entry.

Reply:
Hmmm, so if the user focus two different modules subsequently from the same cell, should we should register only the last one?

Reply:
I'm still trying to figure out the best way to handle that, because when we change the editor focus using `go back` shortcut, it sends to `CursorHistory` to push the new focus.

Reply:
> Hmmm, so if the user focus two different modules subsequently from the same cell, should we should register only the last one?

We can have a heuristic, for example, if the new position is same cell but 10+ lines apart, we push it instead.

> I'm still trying to figure out the best way to handle that, because when we change the editor focus using go back shortcut, it sends to CursorHistory to push the new focus.

When that happens, the focus would be the exact location as the top of the history, so it would be ignored, no?

Reply:
> When that happens, the focus would be the exact location as the top of the history, so it would be ignored, no?

In this case, no, because you could rewrite the future cursor history, when you go back from index 3 to 2 but focuses on another cell that wasn't the same from index 3.

So, we can have both scenarios when changing the focus of the editor :/

Reply:
Given the example below, let's navigate between the cells:

```
{
  entries: [
    { cellId: "123456", line: 1, offset: 13 },
    { cellId: "156162", line: 20, offset: 1 },
    { cellId: "186941", line: 32, offset: 6 },
  ],
  index: 2,
}
```

if we use `goBack()`, we will navigate from index 2 to 1 and the current entry will be `{ cellId: "156162", line: 20, offset: 1 }`, but if we access a new cell, let's say we focus the cell `171416`, it should rewrite and the index 2 will be `{ cellId: "171416", line: 1, offset: 1 }`.

This scenario is fine, but using the implementation when the user focus a different cell, we should capture the last position in the previous cell and save it. But, since I was using the `onBlur` event, when the user change the focus, I was stuck in the push infinitely.

So, trying to use `goBack()` twice was rewriting the stack infinitely. This is the issue I've been trying to fix.

Reply:
So the problem is that we use onblur to push the update, right? What I would do is subscribe to selection changes, so whenever the user moves within the editor, we get an event and already send to the session. When they go to different cell, we don't need to do anything, because the history already has the last position.

Reply:
It works flawlessly